classes/Product.php

```diff
@@ -589,7 +589,7 @@ public function __construct($id_product = null, $full = false, $id_lang = null,
     public function getFieldsShop()
     {
         $fields = parent::getFieldsShop();
-        if (null === $this->update_fields || (!empty($this->update_fields['price']) && !empty($this->update_fields['unit_price']))) {
+        if (isset($this->unit_price) && (null === $this->update_fields || (!empty($this->update_fields['price']) && !empty($this->update_fields['unit_price'])))) {
```

Comment:
```suggestion
        if (isset($this->unit_price) && null === $this->update_fields || (!empty($this->update_fields['price']) && !empty($this->update_fields['unit_price']))) {
```

Reply:
What if `$this->unit_price` is not set, but `$this->update_fields['price']` and `['unit_price']` are set?
`false && false || true` will return `true `and there will be attempt to access `$this->unit_price`